cmd/generator/main.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"os"
+)
+
+func main() {
+	argsWithProg := os.Args
+
+	if len(argsWithProg) != 2 {
+		panic("Wrong amount of arguments!")
+	}
+
+	if len(argsWithProg[1]) == 0 {
+		panic("Given regex is empty!")
+	}
+
+	regex := argsWithProg[1]
+
+	_, err := os.Stdout.WriteString((generateCode(regex) + "\n"))
+	if err != nil {
+		panic("Error writing to stdout")
+	}
+}
+
+func generateCode(regex string) string {
+	afterLastBrace, beforeCurrentBrace, indentations, result := regex, "", "", "rex.New(\n"
+
+	currentOpenBraceIndex, bracesCounter := 0, 0
+
+	for index, runeValue := range regex {
+		if runeValue == '(' {
+			bracesCounter++
+
+			afterLastBrace = ""
+			beforeCurrentBrace = regex[currentOpenBraceIndex:index]
+
+			if len(beforeCurrentBrace) != 0 {
+				indentations += "\t"
+				result = result + indentations + "rex.Common.Raw(`" + beforeCurrentBrace + "`),\n"
+				indentations = indentations[:(len(indentations) - 1)]
+			}
+
+			indentations += "\t"
```

Comment:
Isn't it better to make `indentations` as int, and generate the correct string?

[strings.Repeat](https://pkg.go.dev/strings#Repeat)

It will simplify the code with reslicing.

Reply:
That's a good idea!